### ISSUE TITLE
aax to mp3

### ISSUE THREAD

**Issue by grahamhjones** [openaudible/openaudible#1246]
Converting AAX to MP3 is running extremely slowly. It is running in real time or maybe slightly faster unlike previous times I have used this option.

I am running on Arch linux and using the latest version of OpenAudilble.

**Comment by openaudible** [comment]
AAX -> MP3 can take a while, depending on your CPU and the encoding format. An 9 hour book (500Mb) should encode in a about 5 minutes. But if you have an slow CPU, slow disk drive, or low memory, it can take longer. 

It is a slow process, but we do it as fast as your computer can. 

M4B is super quick because the format doesn't change so there is no decoding and encoding of audio. 

We do convert multiple mp3's at once (when there are multiple books to convert) which can slow things down. 

There may also be a performance issue if you're on a network drive or have a bad or failing hard drive. You may want to run basic diagnostics when the conversion is in process to see if your memory, cpu or disk is getting maxed out.

Lastly, you can look at the openaudible console to see the ffmpeg command we use to convert. You can try running that from the command line, change arguments, or run the same command on a different computer or OS.

**Comment by stratus-ss** [comment]
I also came here seeking the answer to this question. It is REALLY REALLY slow to convert  AAX to MP3 in OpenAudible. As a test, I converted to m4b, then I used audacity to convert the m4b to mp3 on "Extreme" preset.

Book size: 7 hrs 30 minutes
m4b time: 5 seconds
m4b to mp3 via adacity: 1 m 25s

Converting directly with OpenAudible's ffmpeg command: unknown, after 20 minutes it had only converted 34 minutes of the audio book

**Comment by openaudible** [comment]
I know we have several users on Arch so I don’t think that that is a problem. are you on a net work drive?  open Audible uses a temporary dir as well as the library directories so both of those should be fast and accessible. 

You can run the command line tool top. What is the bottleneck? Do you have enough memory? Check your hard drive speeds.

If you shoot me a link to your arch Linux iso  and specs I can test it on a VM.

**Comment by openaudible** [comment]
Yeah it’s the panel that says audiobook info that is blank for you. I don’t Know why. What OS are you on? Send whole screenshot with one book selected. You can redact if you want.

**Comment by stratus-ss** [comment]
Here are the specs of the computer:

```
System:
  Host: stratus-desktop.stratus.local Kernel: 6.1.39-3-lts arch: x86_64
    bits: 64 Desktop: GNOME v: 44.3 Distro: Arch Linux
Machine:
  Type: Desktop System: ASUS product: N/A v: N/A serial: <superuser required>
  Mobo: ASUSTeK model: ROG STRIX B550-A GAMING v: Rev X.0x
Memory:
  System RAM: total: 32 GiB available: 31.23 GiB used: 14.28 GiB (45.7%)
CPU:
  Info: 8-core AMD Ryzen 7 5800X [MT MCP] speed (MHz): avg: 2925
    min/max: 2200/4850
Graphics:
  Device-1: NVIDIA GA104 [GeForce RTX 3060 Ti Lite Hash Rate] driver: nvidia
    v: 535.86.05
Network:
  Device-1: Intel Ethernet I225-V driver: igc
Drives:
  Local Storage: total: 4.58 TiB used: 1.91 TiB (41.8%)
```

The OpenAudible install is on a local NVME drive, a Samsung Evo 970 Plus running XFS (not that that should matter). I am not storing these on a network drive locally. 

When launching the ffmpeg command from the OpenAudible console I can see that it is using only 2-3 of the 16 threads available on the system.

**Comment by stratus-ss** [comment]
> Yeah it’s the panel that says audiobook info that is blank for you. I don’t Know why. What OS are you on? Send whole screenshot with one book selected. You can redact if you want.

sorry which thing do you want me to take a screenshot of?

**Comment by openaudible** [comment]
Sorry I posted the wrong reply about a screenshot. Not sure why it would be slow. Maybe check speeds of disk and check top when converting. You have plenty of memory. 

I can try to reproduce if you send an iso link. 

How about copying files on that computer? Plenty fast?

**Comment by stratus-ss** [comment]
You can find the download link here: https://archlinux.org/download/

As for testing the disk using the following test in the OpenAudible directory:

```
fio --randrepeat=1 --ioengine=libaio --direct=1 --gtod_reduce=1 --name=test --filename=fio-test --bs=4k --iodepth=64 --size=8G --readwrite=randrw --rwmixread=75
```

Shows that this folder has the following:

```
Run status group 0 (all jobs):
   READ: bw=339MiB/s (356MB/s), 339MiB/s-339MiB/s (356MB/s-356MB/s), io=6141MiB (6440MB), run=18097-18097msec
  WRITE: bw=113MiB/s (119MB/s), 113MiB/s-113MiB/s (119MB/s-119MB/s), io=2051MiB (2150MB), run=18097-18097msec
```

This test essentially says create a random file 8G in size with a blocksize of 4k

For reference a Sandisk X400 SSD gets

```
Run status group 0 (all jobs):
   READ: bw=125MiB/s (131MB/s), 125MiB/s-125MiB/s (131MB/s-131MB/s), io=6141MiB (6440MB), run=49014-49014msec
  WRITE: bw=41.8MiB/s (43.9MB/s), 41.8MiB/s-41.8MiB/s (43.9MB/s-43.9MB/s), io=2051MiB (2150MB), run=49014-49014msec
```

So by comparison the OpenAudible (NVME) disk is at least 3 times faster than a standard SSD

**Comment by stratus-ss** [comment]
here is top:

![image](https://github.com/openaudible/openaudible/assets/8162705/96802011-7348-4b5f-ad4a-695fdeb70549)

During this screenshot the book in question is converting at about 5 seconds if book time to every actual second

**Comment by stratus-ss** [comment]
Thanks, I did catch that there were multiple going at the same time. However, this should not (give the size of this system) have been a hindrance. I did purge all of the running jobs, verified that there were no remnants of jobs still running and tested again. It was more-or-less the same

The end result is that if I want an MP3, I need to convert to M4B then convert that to MP3. That is by far the most affective path forward given the apparent inefficiencies in what appears to be AAX conversion by FFMPEG
